Automattic/jetpack
# Update Sync handling of unexpected actions and serialization issues.

Jetpack Sync saves all actions/events into the options table using serialize. However when retrieving the data from the database maybe_unserialize is called that first runs is_serialized before unserializing. This can lead to the unserialized data being sent to WP.com which is unable to process the data. This update adds additional handlers so that any unserialzable data is not sent to WP.com and has additional checks for empty actions that should be discarded.

#### Changes proposed in this Pull Request:
* Reduces # of unknown actions being synced.

#### Jetpack product discussion


N/A

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Apply this PR to your Jetpack site.
* Connect Jetpack to WP.com
* Edit posts, comments, modify content
* Verify that the actions are being synced to WP.com (log2logstash link in Jetpack Debug)
* Run a Sync Checksum via Jetpack Debug and confirm no missing/data
